sql/src/main/java/io/crate/action/sql/TransportSQLAction.java

```diff
@@ -114,7 +115,11 @@ public void pauseProcessed(ResumeHandle resumeable) {
             @Override
             public void finish(RepeatHandle repeatable) {
                 setNextRowResult = RowReceiver.Result.STOP;
-                listener.onResponse(createResponse(analysis, request, rows, startTime));
+                if (analysis.rootRelation() == null) {
```

Comment:
imho, this doesn't look intuitive at first glance. why don't you keep `expectsAffectedRows()` returning the inverse of this boolean condition instead? this way who looks to `Analysis` class will have a clear understanding of it.